pkg/dockerregistry/server/auth.go

```diff
@@ -351,6 +351,14 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				}
 			}
 
+		case "signature":
+			switch access.Action {
+			case "get":
+				// For /signature/<reference> we pass the request to OpenShift API using the user
+				// token so the authorization happen on the OpenShift API side.
+				continue
```

Comment:
I like the idea of handling the authorization for all the routes on one place.

The handler can then safely focus on its job.